### ISSUE TITLE
preparations running twice with mariadb 10.2

### ISSUE THREAD

**Issue by tanner0101** [vapor/fluent#271]
https://github.com/vapor/vapor/issues/1040

**Comment by tanner0101** [comment]
Looks like Fluent is unable to detect that the migration already ran. @janslu I'll attempt to recreate on my computer. Can you provide more information about how you installed MariaDB on both your local and remote machines?

**Comment by tanner0101** [comment]
This also relies on https://github.com/vapor/fluent-provider/issues/24 to get better logs.

**Comment by janslu** [comment]
I'm sorry I wasn't clear - I found no way to use mariadb dev libraries on osx. On homebrew mysql is a dependency for cmysql and I found no way of symlinking to actually build something against mariadb-client. So I was building against mysql 5.7 libraries and connecting to a dockerized 10.2 mariadb server instance (official docker image with 10.2 tag).

On Ubuntu 16.04.2 I have installed the mariadb recommended apt source and installed the server, the client and libmariadb-dev. This last package provides libmysqlclient.so, libmysqlclient.so.18 etc files BUT I needed to create symlink from /usr/lib/libmysqlclient.so and /usr/lib/libmysqlclient.so.18 to /usr/lib/x86_64-linux-gnu/libmysqlclient.so and /usr/lib/x86_64-linux-gnu/libmysqlclient.so.18

It builds ok on my server but errors out the app on each restart. I don't think I can install both libmysqlclient-dev and mariadb-server-10.2 packages at the same time... 

Hope this helps. I will be happy to assist more, I'm a beginner with limited knowledge...

**Comment by tanner0101** [comment]
@janslu try using 

```
-Xswiftc -I/path/to/mariadb/includes -Xlinker -L/path/to/mariadb/libs
```

This is what we use to manually link MySQL:
```
swift build -Xswiftc -I/usr/local/include/mysql -Xlinker -L/usr/local/lib
```

I imagine the same should work if you instead point it to MariaDB and MySQL is not installed.

**Comment by janslu** [comment]
@tanner0101  Thanks for looking into this. I'm sorry I couldn't return earlier - family stuff. I have finally managed to install Ubuntu on vmware for tests, and here's what is/was happening. 
MariaDB 10.2 can (and prefers to) install it's own version of libmysqlclient-dev package, called libmariadb-dev.
I have installed vapor and cmysql earlier and then made an upgrade to mariadb-server-10.2 which caused the client library replacement. This library has problems with a missing symlink and I couldn't build properly until I have added the symlink, but this is a sidenote. 
After vapor is built with libmariadb-dev it errors out with table exists error on each consecutive restart. First start works fine and tables are being prepared. Queries also worked fine. 
cmysql installs "classic" libmysqlclient-dev (version 5.7.xxx, so the library is older then mariadb). With this library the application builds and works fine. No errors on restarts. 
I have installed a test environment (Ubuntu 16.04.2 Server) and I have installed mariadb-server-10.2 successfully, then added vapor and cmysql (which installed libmysqlclient-dev). and everything worked ok. 
I have no idea how to debug this further. I think this may be closed unless I can help you somehow.
